bot/classes/user-role.js

```diff
@@ -42,8 +42,9 @@ class UserRole extends UserRoleModel {
         if (this.discordUserId !== null) {
             return this.discordUserId === userDiscordId;
         } else {
-            if (typeof global.discordClient !== "undefined" && typeof global.discordClient.guilds.get(this.discordGuildId) !== "undefined") {
-                let role = global.discordClient.guilds.get(this.discordGuildId).roles.get(this.discordRoleId);
+            const discordClient = discordClientProvider.get();
+            if (typeof discordClient !== "undefined" && typeof discordClient.guilds.get(this.discordGuildId) !== "undefined") {
```

Comment:
This undefined check is no longer needed, since `.get()` throws when we are offline.